lib/pattern.c

```diff
@@ -51,9 +51,13 @@ static const char *parse_file(const char *beg, char *out,
 	if (fd < 0)
 		goto err_free_out;
 
-	count = read(fd, out, out_len);
-	if (count == -1)
-		goto err_free_close_out;
+	if (out) {
+		count = read(fd, out, out_len);
+		if (count == -1)
+			goto err_free_close_out;
+	} else {
+		count = lseek(fd, 0, SEEK_END);
+	}
```

Comment:
These aren't quite the same, very large reads will not read a whole file in one go

Reply:
Hmm, yes, I guess it'll need to be repeated if it returns short. Though in testing this never happened.

Reply:
I'd probably leave it here, and add a loop on read once the maximum pattern size is increased. with 512-byte patterns I don't think it's an issue.

Reply:
I'd probably leave it here, and add a loop on read once the maximum pattern size is increased. with 512-byte patterns I don't think it's an issue.